### DISCUSSION TITLE
How do WinGet shows progress ring while loading?

### DISCUSSION THREAD

**Discussion by GEOEGII555** [microsoft/winget-cli#2194] [Q&A] [answered]
How winget shows a progress ring while loading?
![изображение](https://user-images.githubusercontent.com/57674559/170833488-d3c0d016-fa45-4ac9-9ab5-5137a6970f6e.png)

**Comment by Masamune3210** [comment]
I would assume the code you are looking for is in AppInstallerProgress.h and ExecutionProgress.h

**Comment by florelis** [comment, marked as answer]
You mean in the Terminal title bar? I believe that (along with, for example, text color) is controlled by outputting special character sequences that the Terminal interprets in specific ways. In winget, these are generated in [VTSupport.cpp](https://github.com/microsoft/winget-cli/blob/3f4645ae09be2c304f25e33a7642c65de432e897/src/AppInstallerCLICore/VTSupport.cpp). There's some documentation about these special sequences in [Console Virtual Terminal Sequences](https://docs.microsoft.com/windows/console/console-virtual-terminal-sequences)
